### PR TITLE
Initialize parser lazily

### PR DESCRIPTION
**Pull request by masatake**
* Make "units" a bit faster.
* Regex patterns and keywords can be used declarative way.

### CODE REVIEW

File: main/parse.h [hunk 1]
```diff
@@ -67,6 +79,10 @@ typedef struct {
 	unsigned int method;           /* See PARSE__... definitions above */
 	boolean useCork;
 	boolean allowNullTag;
+	tagRegexTable *tagRegexTable;
```
Comment (b4n) [inline main/parse.h]: elements should probably be const

File: main/parse.h [hunk 2]
```diff
@@ -67,6 +79,10 @@ typedef struct {
...
+	unsigned int tagRegexCount;
+	keywordTable *keywordTable;
```
Comment (b4n) [inline main/parse.h]: ditto

File: parsers/ant.c [hunk 3]
```diff
@@ -15,22 +15,21 @@
 #include <string.h>
 #include "parse.h"
 
+static tagRegexTable antTagRegexTable [] = {
```
Comment (b4n) [inline parsers/ant.c]: this and other tables should probably be all static

Comment (b4n) [inline parsers/ant.c]: sorry, s/static/const/

**Comment by b4n** [comment]
I didn't really check the initialization logic changes, but using tables looks like a nice idea.

**Comment by masatake** [comment]
@b4n, thank you. I added a new commit where I reflected your comment.

**Comment by vhda** [comment]
Is keeping `KIND_COUNT` name so incorrect? Because if we kept it the number of modified files would decrease drastically.

**Comment by masatake** [comment]
@vhda, CURRENTLY I don't want to provide COUNT macros for each data type: KIND_COUNT, REGEX_COUNT, FIELD_COUNT, KEYWORD_COUNT, and FLAG_COUNT. 

Defining macros for each type itself is not bad. However, I don't want to do it now because the header files of ctags are not well organized. When we can split the headers file for each data type, I would like to introduce COUNT macros for each data type.

**Comment by vhda** [comment]
Understood and it makes sense.
Thanks! :)

**Comment by masatake** [comment]
O.k. I will merge this. This makes units faster. 

I really need these commits because I'm using i386(http://ascii.jp/elem/000/000/214/214977/). It takes really long time to run units.

**Pull request merged**